tests/dnsforwardzone/test_dnsforwardzone.yml

```diff
@@ -0,0 +1,136 @@
+---
+- name: Tests
+  hosts: ipaserver
+  become: true
+  gather_facts: false
+
+  tasks:
+  - name: ensure forwardzone example.com is absent
+    ipadnsforwardzone:
+      ipaadmin_password: password01
+      name: example.com
+      ensure: absent
+
+  - name: ensure forwardzone example.com is created
+    ipadnsforwardzone:
+      ipaadmin_password: password01
+      state: present
+      name: example.com
+      forwarders:
+        - 8.8.8.8
+      forwardpolicy: first
+      skip_overlap_check: true
+    register: result
+    failed_when: not result.changed
+
+  - name: ensure forwardzone example.com is present again
+    ipadnsforwardzone:
+      ipaadmin_password: password01
+      state: present
+      name: example.com
+      forwarders:
+        - 8.8.8.8
+      forwardpolicy: first
+      skip_overlap_check: true
+    register: result
+    failed_when: result.changed
+
+  - name: ensure forwardzone example.com has two forwarders
+    ipadnsforwardzone:
+      ipaadmin_password: password01
+      state: present
+      name: example.com
+      forwarders:
+        - 8.8.8.8
+        - 4.4.4.4
+      forwardpolicy: first
+      skip_overlap_check: true
+    register: result
+    failed_when: not result.changed
+
+  - name: ensure forwardzone example.com has one forwarder again
+    ipadnsforwardzone:
+      ipaadmin_password: password01
+      state: present
+      name: example.com
+      forwarders:
+        - 8.8.8.8
+      forwardpolicy: first
+      skip_overlap_check: true
+      ensure: present
+    register: result
+    failed_when: not result.changed
+
+  - name: skip_overlap_check can only be set on creation so do not change when its changed 
+    ipadnsforwardzone:
+      ipaadmin_password: password01
+      state: present
+      name: example.com
+      forwarders:
+        - 8.8.8.8
+      forwardpolicy: first
+      skip_overlap_check: false 
+      ensure: present
+    register: result
+    failed_when: result.changed
+
+  - name: change all the things at once
+    ipadnsforwardzone:
+      ipaadmin_password: password01
+      state: present
+      name: example.com
+      forwarders:
+        - 8.8.8.8
+        - 4.4.4.4
+      forwardpolicy: only
+      skip_overlap_check: false 
+      ensure: present
```

Comment:
...

Reply:
You are using `state` and `ensure` here.. that results in a warning: `[WARNING]: Both option state and its alias ensure are set.`